pkg/apis/monitoring/v1beta1/alertmanager_config_types.go

```diff
@@ -863,6 +865,23 @@ type TelegramConfig struct {
 	HTTPConfig *HTTPConfig `json:"httpConfig,omitempty"`
 }
 
+type MSTeamsConfig struct {
+	// Whether to notify about resolved alerts.
+	// +optional
+	SendResolved *bool `json:"sendResolved,omitempty"`
+	// MSTeams webhook url
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern:="^http(s)?://.+$"
+	WebhookUrl string `json:"webhookUrl"`
+	// +optional
+	Title string `json:"title,omitempty"`
```

Comment:
If those are optional should it be a pointer `*string`?

Reply:
sorry not sure why do we have to use pointer instead of string here? suppose they're only used for crd version conversion.

Reply:
See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required :)